web/models.py

```
@@ -2787,3 +2787,105 @@ class ScheduledPost(models.Model):
+        elif self.node_type == "course" and self.enrollment:
+            if hasattr(self.enrollment, "progress"):
+                return self.enrollment.progress.completion_percentage
+            return 0
+
+        elif self.node_type == "milestone" and self.target_value:
+            return min(100, int((self.current_value / self.target_value) * 100)) if self.target_value else 0
+
+        return 0
+
+    @property
+    def is_completed(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `is_completed`

(ANN201)